packages/flutter_test/lib/src/window.dart

```diff
@@ -447,13 +425,11 @@ class TestWindow implements ui.SingletonFlutterWindow {
   }
 
   @override
-  ui.ViewConfiguration get viewConfiguration => _viewConfiguration ?? _window.viewConfiguration;
-  ui.ViewConfiguration? _viewConfiguration;
-
+  ui.ViewConfiguration get viewConfiguration => _view.viewConfiguration;
   /// Hide the real view configuration and report the provided [value] instead.
-  set viewConfigurationTestValue(ui.ViewConfiguration? value) { // ignore: avoid_setters_without_getters
-    _viewConfiguration = value;
-    onMetricsChanged?.call();
+  // ignore: avoid_setters_without_getters
+  set viewConfigurationTestValue(ui.ViewConfiguration? value) {
+    _view._viewConfigurationLegacy = value;
```

Comment:
Oh, this is likely the reason View and Window have to stay in the same file? Maybe we should move them both to view.dart then...

Reply:
Or, you could just make viewConfigurationLegacy and mark it as deprecated right away. Not as clean, but we have done that before...

Reply:
I think, per Hixie's comment in chat, that people shouldn't really be pointing directly to this _file_. Instead, they'll be pointing to `package:flutter_test/flutter_test.dart`.

I agree that when we get rid of `TestWindow`, the other classes should move into different files, but I don't think it'll count as a breaking change when we do, so I'm not to worried about it right now since it lets us handle this particular case gracefully.

Reply:
Also, if it _does_ count as a breaking change, the `Deprecated` attribute does work on libraries. So we can move the classes to a new file and re-export them here for the backwards compatibility window. If worse comes to worse.